test.py

```diff
@@ -414,7 +414,17 @@ def __init__(self, path, cfg: dict, options: argparse.Namespace, mode: str) -> N
 
         cluster_cfg = self.cfg.get("cluster", {"initial_size": 1})
         cluster_size = cluster_cfg["initial_size"]
-        pool_size = cfg.get("pool_size", 2)
+
+        env_pool_size = os.getenv("CLUSTER_POOL_SIZE")
+        if env_pool_size is not None:
+            pool_size = env_pool_size
+        elif options.cluster_pool_size is not None:
+            pool_size = options.cluster_pool_size
+        else:
```

Comment:
I think the ordering is wrong. First priority should be to the command-line parameter, then the environment variable, then the default.

Reply:
(could have been written: pool_size = options.cluster_pool_size or os.getenv(...) or cfg.get(...))